infra/stream/Formatter.cpp

```diff
@@ -0,0 +1,225 @@
+#include "infra/stream/Formatter.hpp"
+#include "infra/stream/StreamErrorPolicy.hpp"
+#include <cctype>
+
+namespace infra
+{
+    FormatSpec::FormatSpec(const char*& format)
+    : format(format)
+    {
+        if (*format != ':')
+            return;
+        ++format;
+        ParseAlign();
+        ParseZero();
+        ParseWidth();
+        ParseType();
+    }
+
+    void FormatSpec::ParseAlign()
+    {
+        auto i = (*format == '\0') ? 0 : 1;
+        do
+        {
+            switch (format[i])
+            {
+            case '<':
+                align = FormatAlign::left;
+                break;
+            case '^':
+                align = FormatAlign::center;
+                break;
+            case '>':
+                align = FormatAlign::right;
+                break;
+            default:
+                break;
+            }
+            if (align != FormatAlign::nothing)
+            {
+                if (i == 1)
+                    fill = *format++;
+                ++format;
+                return;
+            }
+        } while (i-- > 0);
+    }
+
+    void FormatSpec::ParseZero()
+    {
+        if (*format != '0')
+            return;
+        if (align == FormatAlign::nothing)
+            fill = '0';
+        format++;
+    }
+
+    void FormatSpec::ParseWidth()
+    {
+        while (isdigit(*format))
```

Comment:
I think you are missing a cast. See https://en.cppreference.com/w/cpp/string/byte/isdigit.